### ISSUE TITLE
warning: 'this' pointer is null

### ISSUE THREAD

**Issue by xantares** [PDAL/PDAL#3517]
at filters/private/expr/ConditionalParser.cpp:288
the code uses a null pointer in the node variable node in the else branch:
```
ConstValueNode *node = dynamic_cast<ConstValueNode *>(sub.get());
    if (node)
    {
        double v = -(node->value());
        expr.pushNode(NodePtr(new ConstValueNode(v)));
    }
    else
    {
        if (node->isBool())
        {
            setError("Can't apply '-' to logical expression '" +
                sub->print() + "'.");
            return false;
        }
        expr.pushNode(NodePtr(new UnMathNode(NodeType::Negative, std::move(sub))));
    }
```

the same issue is found in filters/private/expr/MathParser.cpp:140

**Comment by abellgithub** [comment]
Thanks for this report.

**Comment by abellgithub** [comment]
Fixed by #3595
